scripts/pythondistdeps.py

```diff
@@ -112,76 +118,39 @@ def convert_ordered(name, operator, version_id):
         version = RpmVersion(version_id)
     return '{} {} {}'.format(name, operator, version)
 
+
 OPERATORS = {'~=': convert_compatible,
              '==': convert_equal,
              '===': convert_arbitrary_equal,
              '!=': convert_not_equal,
              '<=': convert_ordered,
-             '<':  convert_ordered,
+             '<': convert_ordered,
              '>=': convert_ordered,
-             '>':  convert_ordered}
+             '>': convert_ordered}
+
 
 def convert(name, operator, version_id):
     return OPERATORS[operator](name, operator, version_id)
 
 
-opts, args = getopt(
-    argv[1:], 'hPRrCEMmLl:',
-    ['help', 'provides', 'requires', 'recommends', 'conflicts', 'extras', 'majorver-provides', 'majorver-only', 'legacy-provides' , 'legacy'])
-
-Provides = False
-Requires = False
-Recommends = False
-Conflicts = False
-Extras = False
-Provides_PyMajorVer_Variant = False
-PyMajorVer_Deps = False
-legacy_Provides = False
-legacy = False
+parser = argparse.ArgumentParser(prog=argv[0])
+parser.add_argument('-P', '--provides', action='store_true', help='Print Provides')
+parser.add_argument('-R', '--requires', action='store_true', help='Print Requires')
+parser.add_argument('-r', '--recommends', action='store_true', help='Print Recommends')
+parser.add_argument('-C', '--conflicts', action='store_true', help='Print Conflicts')
+parser.add_argument('-E', '--extras', action='store_true', help='Print Extras')
```

Comment:
I tihnk this should be moved to its own mutually exclusive group.

Reply:
Ahh, now getopt didn't do that! :-P But sure, let me look at doing that.

Reply:
Sure, but since we are now doing refactor of this code, it makes sense to implement things like this so it is not possible to pass bunch of parameters which together do not make sense.